### PR TITLE
Parse semgrep expression in Rust parser

### PR DESCRIPTION
**Pull request by Ruin0x11**
Adds semgrep patterns and an alternate toplevel entry point to the Rust parser.

### CODE REVIEW

File: semgrep-core/tests/Test.ml [hunk 1]
```diff
@@ -258,6 +258,12 @@ let lang_regression_tests =
     let lang = Lang.Lua in
     regression_tests_for_lang files lang
   );
+  "semgrep Rust" >::: (
+    let dir = Filename.concat tests_path "rust" in
+    let files = Common2.glob (spf "%s/*.rs" dir) in
```
Comment (aryx) [inline semgrep-core/tests/Test.ml:263]: Did you forget to add some test files in the PR?
The glob might raise an exn if there is no file in the directory I think.

**Review by aryx — APPROVED** [review]
Could you remove your modif on pfff as it generates a conflicts (and the modification shown have already been merged).

**Review by aryx — CHANGES_REQUESTED** [review]
Remove modif to pfff so I can merge.

**Comment by aryx** [comment]
Not sure why, but above on the "This branch has conflicts that must be resolved" I can't even click on the "Use the command line" link; it does not show any help anymore (it used to show a list of git commands to run to fix the conflict).
Github bug?

**Comment by brendongo** [comment]
It probably has something to do with pfff being a submodule?

**Comment by aryx** [comment]
Maybe. I thought I had similar conflicts before and was able to click on "use the command line" but maybe it was never a conflict about a submodule.

File: semgrep-core/tests/Test.ml (hunk 1 above)
Comment (Ruin0x11) [reply semgrep-core/tests/Test.ml:263]: Yes, I did. I added them.

**Review by aryx — APPROVED** [review]

**Pull request merged**